var/spack/repos/builtin/packages/wannier90/package.py

```diff
@@ -61,10 +89,82 @@ def edit(self, spec, prefix):
+                             'dos.o', 'berry.o', 'kpath.o', 'kslice.o',
+                             'boltzwann.o', 'geninterp.o']
+                new_objs_post = ['comms.o', 'get_oper.o', 'postw90_common.o',
+                                 'wan_ham.o', 'spin.o', 'dos.o', 'berry.o',
+                                 'kpath.o', 'kslice.o', 'boltzwann.o',
+                                 'geninterp.o']
+                filter_file(''.join(['OBJS_POST  = ', ' '.join(objs_post)]),
+                            ''.join(['OBJS_POST  = ',
+                                     ' '.join(new_objs_post)]),
+                            join_path(self.stage.source_path,
+                                      'src/Makefile.2'))
+                filter_file('../../wannier90.x: .*',
+                            ' '.join(['../../wannier90.x: $(OBJS)',
+                                      '../wannier_prog.F90 $(LIBRARY)']),
+                            join_path(self.stage.source_path,
+                                      'src/Makefile.2'),
+                            string=True)
+                filter_file(' '.join(['../../postw90.x: $(OBJS_POST)',
+                                      '$(POSTDIR)postw90.F90']),
+                            ' '.join(['../../postw90.x: $(OBJS_POST)',
+                                     '$(POSTDIR)postw90.F90 $(LIBRARY)']),
+                            join_path(self.stage.source_path,
+                                      'src/Makefile.2'), string=True)
+                filter_file(
+                    ' '.join([
+                        '$(COMPILER) ../wannier_prog.F90',
+                        '$(LDOPTS) $(OBJS) $(LIBS)',
+                        '-o ../../wannier90.x']),
+                    ' '.join(['$(COMPILER) -I../obj ../wannier_prog.F90',
+                              '$(LDOPTS) -L../.. -lwannier',
+                              '-o ../../wannier90.x']),
+                    join_path(self.stage.source_path,
+                              'src/Makefile.2'), string=True)
+                filter_file(
+                    ' '.join(['$(COMPILER) $(POSTDIR)postw90.F90',
+                              '$(POSTOPTS) $(LDOPTS)',
+                              '$(OBJS_POST)',
+                              '$(LIBS) -o ../../postw90.x']),
+                    ' '.join(['$(COMPILER) -I../obj $(POSTDIR)postw90.F90',
+                              '$(POSTOPTS) $(LDOPTS) $(OBJS_POST)',
+                              '-L../.. -lwannier $(LIBS) -o ../../postw90.x']),
+                    join_path(self.stage.source_path,
+                              'src/Makefile.2'), string=True)
+
+                filter_file(
+                    ' '.join(['$(AR) $(ARFLAGS)',
+                              '$(LIBRARY) $(OBJS2) $(OBJS)']),
+                    ' '.join(['$(MPIF90) $(FCOPTS) -shared -o '
+                              '$(LIBRARY) $(OBJS2) $(OBJS) $(LIBS)']),
```

Comment:
I'm pretty sure you can do something like this:
```suggestion
                    '$(AR) $(ARFLAGS) '
                    '$(LIBRARY) $(OBJS2) $(OBJS)',
                    '$(MPIF90) $(FCOPTS) -shared -o '
                    '$(LIBRARY) $(OBJS2) $(OBJS) $(LIBS)',
```
Try that out and see if it parses.

Reply:
It looks like it does!

Reply:
Can you do this for the other `' '.join([...])` usages too?